app/presenters/hyrax/batch_ingest/batch_item_presenter.rb

```diff
@@ -37,42 +43,39 @@ def repo_object_display_name
         repo_object_class&.model_name&.human || repo_object_class_name
       end
 
+      # Memoized results of BatchItem#repo_object_exists?
+      def repo_object_exists?
+        @repo_object_exists ||= batch_item.repo_object_exists?
+      end
+
       class << self
         def status_labels
           {
             initialized: I18n.t('hyrax.batch_ingest.batch_items.status.initialized'),
             enqueued:    I18n.t('hyrax.batch_ingest.batch_items.status.enqueued'),
             running:     I18n.t('hyrax.batch_ingest.batch_items.status.running'),
             completed:   I18n.t('hyrax.batch_ingest.batch_items.status.completed'),
-            failed:      I18n.t('hyrax.batch_ingest.batch_items.status.failed'),
-            expunged:    I18n.t('hyrax.batch_ingest.batch_items.status.expunged')
-          }.with_indifferent_access
-        end
-
-        def status_css_classes
-          {
-            initialized:  'fa fa-info',
-            enqueued:     'fa fa-info',
-            running:      'fa fa-refresh fa-sync',
-            completed:    'fa fa-check-circle',
-            failed:       'fa fa-exclamation-triangle',
-            expunged:     'fa fa-exclamation-triangle'
+            failed:      I18n.t('hyrax.batch_ingest.batch_items.status.failed')
           }.with_indifferent_access
         end
       end
 
       private
 
         def repo_object_class
-          Object.const_get(batch_item.repo_object_class_name.to_s)
+          Object.const_get(repo_object_class_name.to_s)
         rescue NameError
           nil
         end
 
         def repo_object_path_helper
-          return nil unless repo_object_class
-          self.class.include Rails.application.routes.url_helpers
-          "hyrax_#{repo_object_class.model_name.singular}_path"
+          @repo_object_path_helper ||= begin
+            if repo_object_class
+              self.class.include Rails.application.routes.url_helpers
+              path_helper = "hyrax_#{repo_object_class.model_name.singular}_path".to_sym
+              path_helper if respond_to? path_helper
```

Comment:
What should this return if `respond_to? path_helper` returns false?  I think it will still return path_helper as currently written since it was the last returned value and line 76 would conditionally not run.

Reply:
I don't thinks so... unless i'm missing what you're saying. Consider the example:
```
def foo
  "first value"
  "second value" if false
end
=> :foo 
foo
=> nil
```

Reply:
👍 Cool, thanks for clearing this up.